electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,401 @@
+            else:
+                coin_network = bitbox02.btc.TBTC
+                out_type = bitbox02.btc.BTCPubRequest.UPUB
+        elif xtype == "standard":
+            coin_network = bitbox02.btc.TBTC
+            out_type = bitbox02.btc.BTCPubRequest.TPUB
+        else:
+            raise Exception("invalid keypath xtype")
+
+        xpub = self.device.btc_pub(
+                keypath = xpub_keypath,
+                output_type = out_type,
+                coin = coin_network,
+                display = False, # do not display xpub
+            )
+        print(xpub)
+        return xpub
+
+    def is_pairable(self):
+        # the bitbox should already be setup
+        if not self.device.device_info()["initialized"]:
+            return False
+        return True
+
+    def show_address(self, bip32_path, address_type):
+        print("getting address on: ", address_type, bip32_path)
+        address_keypath, is_mainnet = self.bip32_list_from_keypath_string(bip32_path)
+        if address_type == "p2wpkh" or address_type == "p2pkh" or address_type == "standard":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                script = bitbox02.btc.SCRIPT_P2WPKH
+            else:
+                coin_network = bitbox02.btc.TBTC
+                script = bitbox02.btc.SCRIPT_P2WPKH
+        elif address_type == "p2wpkh-p2sh":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                script = bitbox02.btc.SCRIPT_P2WPKH_P2SH
+            else:
+                coin_network = bitbox02.btc.TBTC
+                script = bitbox02.btc.SCRIPT_P2WPKH_P2SH
+        else:
+            raise Exception("invalid keypath address_type")
+
+        address = self.device.btc_pub(
+                keypath = address_keypath,
+                output_type = bitbox02.btc.BTCPubRequest.ADDRESS,  # pylint: disable=no-member
+                script_type = script,
+                coin = coin_network,
+            )
+        return address
+
+    def sign_transaction(self, keystore, tx):
+        if tx.is_complete():
+            return
+
+        inputhasharray = []
+        hasharray = []
+        pubkeyarray = []
+        network = bitbox02.btc.BTC
+        if constants.net.TESTNET:
+            network = bitbox02.btc.TBTC
+
+        tx_script_type = bitbox02.btc.SCRIPT_P2WPKH
+
+        # Build BTCInputType list
+        inputs = []
+        for txin in tx.inputs():
+            if txin.is_coinbase():
+                self.give_error("Coinbase transactions are not supported")
+            _, full_path = keystore.find_my_pubkey_in_txinout(txin)
+
+            inputs.append({
+                "prev_out_hash": txin.prevout.txid,
+                "prev_out_index": txin.prevout.out_idx,
+                "prev_out_value": txin.value_sats(),
+                "sequence": txin.nsequence,
+                "keypath": full_path,
+            })
+
+            if full_path[0] == 49 + HARDENED:
```

Comment:
what about using `txin.script_type` to map the electrum input script type to bitbox02 script type?

Reply:
also should check that all inputs are the same type. or start with `tx_script_type = None`, set it here if None, otherwise compare that it must be equal.